scipy/signal/windows/_windows.py

```diff
@@ -14,6 +15,13 @@
            'dpss', 'get_window', 'lanczos']
 
 
+def _get_array_namespace(array_namespace):
+    if array_namespace is None:
+        return np_compat
```

Comment:
In fft, we used the pattern `xp = np if xp is None else xp`. There is an argument to be made that the returned namespace should always be array API compatible. However, it is probably less disruptive to keep returning plain `np` for now. That probably makes more sense as a default, so that users aren't unexpectedly getting arrays from `array_api_compat` when they haven't enabled any array API stuff. Thoughts?

Reply:
I did that originally, but found that there are many array API functions that are not available in numpy 1.X (e.g. `np.cumulative_sum`, `np.acos`, np.acosh`, `np.concat`.

Originally I checked for these via `hasattr`, but then found the `np_compat` which seems to take care of these issues already.

I thought that `np_compat` was a cleaner solution than using `xp = np` and then overriding a bunch of things via `hasattr`, but I'm happy to switch back if you think the other approach would be cleaner.

Reply:
Maybe I'm misunderstanding what `np_compat` is: I thought it was `numpy` with some extra function definitions. Under what circumstance would these operations not result in a numpy array?

Reply:
No, I think I'm the one getting confused (it's easy with the complicated history involving `numpy.array_api`...). Since we are just returning arrays here, and the namespace used is an internal implementation detail, we are fine to use `np_compat` 👍. Resolving (maybe we would want to move this helper to `scipy/_lib/_array_api.py`).